x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyUtils.java

```diff
@@ -35,32 +36,39 @@ public class LifecyclePolicyUtils {
     private LifecyclePolicyUtils() {};
 
     /**
-     * Loads a built-in index lifecycle policy and returns its source.
+     * Loads a built-in index lifecycle policy and returns its source with the given parser.
      */
     public static LifecyclePolicy loadPolicy(
         String name,
         String resource,
         Map<String, String> variables,
-        NamedXContentRegistry xContentRegistry
+        NamedXContentRegistry xContentRegistry,
+        XContentType contentType
     ) {
         try {
             String source = TemplateResources.load(resource);
             source = replaceVariables(source, variables);
             validate(source);
 
-            try (
-                XContentParser parser = XContentType.JSON.xContent()
-                    .createParser(XContentParserConfiguration.EMPTY.withRegistry(xContentRegistry), source)
-            ) {
-                LifecyclePolicy policy = LifecyclePolicy.parse(parser, name);
-                policy.validate();
-                return policy;
-            }
+            return parsePolicy(source, name, xContentRegistry, contentType);
         } catch (Exception e) {
             throw new IllegalArgumentException("unable to load policy [" + name + "] from [" + resource + "]", e);
         }
     }
 
+    public static LifecyclePolicy parsePolicy(
```

Comment:
Since this is public, can you add javadocs please? (mention that this method does _not_ do the variable substitution, that is the caller's responsibility)